lib/spack/spack/test/installer.py

```diff
@@ -510,6 +510,65 @@ def test_install_failed(install_mockery, monkeypatch, capsys):
     assert 'Warning: b failed to install' in out
 
 
+def test_install_fail_on_interrupt(install_mockery, monkeypatch, capsys):
+    """Test ctrl-c interrupted install."""
+    err_msg = 'mock keyboard interrupt'
+
+    def _interrupt(installer, task, **kwargs):
+        raise KeyboardInterrupt(err_msg)
+
+    spec, installer = create_installer('a')
+
+    # Raise a KeyboardInterrupt error to trigger early termination
+    monkeypatch.setattr(inst.PackageInstaller, '_install_task', _interrupt)
+
+    with pytest.raises(KeyboardInterrupt, match=err_msg):
+        installer.install()
+
+    out = str(capsys.readouterr())
+    assert 'Failed to install' in out
+    assert err_msg in out
+
+
+def test_install_fail_fast_on_detect(install_mockery, monkeypatch, capsys):
+    """Test fail_fast install when detect an install failure."""
+    spec, installer = create_installer('a')
+
+    # Make sure the package is identified as failed
+    monkeypatch.setattr(spack.database.Database, 'prefix_failed', _true)
+
+    # Skip the actual installation though it should never get there
+    monkeypatch.setattr(inst.PackageInstaller, '_install_task', _noop)
+
+    msg = 'Fail-fast termination of install'
+    with pytest.raises(spack.installer.InstallError, match=msg):
+        installer.install(fail_fast=True)
+
+    out = str(capsys.readouterr())
+    assert 'b failed to install' in out
```

Comment:
True, but it does ensure you are getting the output you expect.

Reply:
My opinion is that it isn't critical what *exactly* gets printed. If this corresponds somehow to the inner state of the object, I'd be inclined to argue we ought to make that measurable without checking what is printed (since the message can move around).

(For command-level tests, I think it can be important to check for output, but for object tests, I think it either provides false assurance or fragility)

Reply:
Removed check of `InstallError` message.

Reply:
This check still appears to be here. Or were you referring to removing another piece of code?

Reply:
I was referring to removal of one of the messages.

Reply:
IMHO I should've been checking the message that stated the installation of `a` was being skipped since it was more meaningful so my latest (forced) push makes that change.